portal/src/graphql/portal/ApplicationsConfigurationScreen.tsx

```diff
@@ -83,33 +93,87 @@ function makeOAuthClientListColumns(
       minWidth: 300,
       className: styles.columnHeader,
     },
-    { key: "action", name: renderToString("action"), minWidth: 200 },
+    {
+      key: "action",
+      name: renderToString("action"),
+      className: styles.columnHeader,
+      minWidth: 200,
+    },
   ];
 }
 
+interface OAuthClientIdCellProps {
+  clientId: string;
+  dismissTimeout?: number;
+}
+
+const OAuthClientIdCell: React.FC<OAuthClientIdCellProps> =
+  function OAuthClientIdCell(props) {
+    const { clientId, dismissTimeout } = props;
+    const copyButtonId = "oauth-client-id-copy-button";
+    const [isCalloutVisible, setIsCalloutVisible] = useState(false);
+    const [calloutToggle, setCalloutToggle] = useState(false);
+    const { renderToString } = useContext(Context);
+
+    const handleCalloutDismiss = () => setIsCalloutVisible(false);
+
+    useEffect(() => {
+      setIsCalloutVisible(true);
+      const timer = setTimeout(handleCalloutDismiss, dismissTimeout ?? 2000);
+      return () => clearTimeout(timer);
+    }, [calloutToggle, dismissTimeout]);
+
+    const onCopyClick = () => {
+      copyToClipboard(clientId);
+      setCalloutToggle((v) => !v);
+    };
+
+    return (
+      <>
+        <span className={styles.cellContent}>{clientId}</span>
+        <IconButton
+          id={copyButtonId}
+          onClick={onCopyClick}
+          iconProps={{ iconName: "Copy" }}
+          title="Copy"
+          ariaLabel="Copy"
+          styles={{ root: styles.copyButtonRoot }}
```

Comment:
This object can be lifted as module-level constant.

Reply:
Can we add 1 more condition to dismiss the callout? The condition is after 0.5s when onMouseLeave fires on IconButton.

Can we tidy up the show/dismiss logic as a hook?